src/query/expression/src/utils/display.rs

```diff
@@ -1073,9 +1073,5 @@ fn display_f64(num: f64) -> String {
 
 /// Display a tuple field name, if it contains uppercase letters or special characters, add quotes.
 pub fn display_tuple_field_name(field_name: &str) -> String {
-    if field_name.chars().any(|c| c.is_ascii_uppercase()) || ident_needs_quote(field_name) {
-        QuotedIdent(field_name, '"').to_string()
-    } else {
-        field_name.to_string()
-    }
+    display_ident(field_name, true, Dialect::PostgreSQL)
```

Comment:
@b41sh It would be better if we can get the actual dialect and `quoted_ident_case_sensitive`